library/std/src/sys/args/unsupported.rs

```diff
@@ -15,22 +15,28 @@ impl fmt::Debug for Args {
 
 impl Iterator for Args {
     type Item = OsString;
+
+    #[inline]
     fn next(&mut self) -> Option<OsString> {
         None
     }
+
+    #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
         (0, Some(0))
     }
 }
```

Comment:
what, no `is_empty` optimization?

Reply:
I kid, I kid

Reply:
Haha, I did consider adding more impls for unsupported, but, yeah, it's so easy to be automatically optimized.

Reply:
yeah. and if anyone with an args-stubbing OS wants to complain about the optimization quality of `Args` iteration they can hand-roll the optimizations themselves.